### DISCUSSION TITLE
How to create sitemaps at buildtime instead of runtime.

### DISCUSSION THREAD

**Discussion by Chirag-dev9** [iamvishnusankar/next-sitemap#571] [Q&A] [answered]
As mentioned here, https://github.com/iamvishnusankar/next-sitemap/discussions/544#discussioncomment-4440491, the postbuild command is not required for serverside sitemap generation, which means the dynamic data will be generated at Runtime when crawlers visit the sitemap page.

So is there a way to generate sitemap at **build time**, so API call is made when building and whole xml file with all urls in generated at build time?

**Q2)** Is it possible to define in build script if I only want to generate one specific sitemap like **my-custom-sitemap-2.xml** and not all the sitemaps?

**Comment by iamvishnusankar** [comment, marked as answer]
`So is there a way to generate sitemap at build time, so API call is made when building and whole xml file with all urls in generated at build time?`

https://github.com/iamvishnusankar/next-sitemap#additional-paths-function

`Q2) Is it possible to define in build script if I only want to generate one specific sitemap like my-custom-sitemap-2.xml and not all the sitemaps?`

Yes, you can set the sitemapBaseFileName and disable generation of index sitemap.

**Reply by Chirag-dev9** [reply]
But that's just one config file with one additional-paths-function . I have multiple sitemaps.
How do I define which urls get added to which sitemap?

I have three different sitemaps and I want to generate them at "next build" with data I get from CMS.
